ui-templates/README.md

```diff
@@ -0,0 +1,52 @@
+# UI Templates
+These JSON objects define templates for capturing user data from a UI and using it to set the parameters of SimulariumIO objects in order to visualize simulation trajectories.
+
+
+## Data types
+Different types of data need to be collected. Some of the data types are "base" HTML Form Input types, and some are "custom" types that contain parameters of the base types or other custom types. Some custom types are "main" types for visualizing data from a specific simulation engine.
+
+Each custom data type has:
+* `python::module` - Python module
+* `python::object` - Python class for the object to populate
+* `parameters` - a dict of names of members of the Python class mapped to information about how to set them. For each:
+  * `name` - the name to display next to the UI Input element
+  * `data_type` - the data type to collect, either a base or custom data type
+  * `description` - instructions for the user about how to fill this field
+  * `required` - is this field required in order to submit?
+  * `default` - the default value to use if none is provided by user, if not provided for a required field, the default is `None`
+  * `match` - a regex pattern to validate the data against (optional)
+  * `help` - additional information to help the user, including what default value will be used (optional)
+  * some types also have additional parameter information
+
+### Main types
+* `smoldyn_data` - information for populating a `SmoldynData` to visualize a trajectory from Smoldyn
+* (more coming soon)
+
+### Base types
+
+* `float`
+* `text`
+* `url`
+* `color`
+
+### Custom types
+
+* `meta_data`
```

Comment:
do you intend to document these other fields here?  some have notes and some don't

Reply:
for example, what's the difference between meta_data and model_meta_data?